### ISSUE TITLE
`max_number_of_steps` ignored in Distributed Layout?

### ISSUE THREAD

**Issue by wcarvalho** [deepmind/acme#198]
It looks like the argument `max_number_of_steps` is ignored or has unexpected behavior in `DistributedLayout`?

I set the number of 10 million and the algorithm runs for well over (currently at 50 million). I assumed that `max_number_of_steps` specifies how many environment steps to run an algorithm before terminating everything (actors, evaluators, learner, etc.). Is that incorrect?

**Comment by ethanluoyc** [comment]
From a look at distributed_layout, it does look like it used to create a coordinator that knows how to terminate the program. Are you using one of the distributed agents in Acme and getting this behavior?

**Comment by wcarvalho** [comment]
I am using a a class that's derivative of `R2D2`. The only change I made was to overwrite `make_learner` for a custom loss function so it should behave as normal?

This is what the new builder looks like. I am using the [learning_lib](https://github.com/deepmind/acme/blob/master/acme/agents/jax/dqn/learning_lib.py) from the DQN Jax agent. Maybe that breaks something?

```
class CustomBuilder(r2d2.R2D2Builder):
  """CustomBuilder agent Builder. Agent is derivative of R2D2 but may use different network/loss function
  """
  def __init__(self,
               networks: r2d2_networks.R2D2Networks,
               config: r2d2_config.R2D2Config,
               logger_fn: Callable[[], loggers.Logger] = lambda: None,
               LossFn: learning_lib.LossFn=R2D2Learning,
               LossFnKwargs=None):
    super().__init__(networks=networks, config=config, logger_fn=logger_fn)

    LossFnKwargs = LossFnKwargs or dict()
    self.loss_fn = LossFn(**LossFnKwargs)

  def make_learner(
      self,
      random_key: networks_lib.PRNGKey,
      networks: r2d2_networks.R2D2Networks,
      dataset: Iterator[reverb.ReplaySample],
      replay_client: Optional[reverb.Client] = None,
      counter: Optional[counting.Counter] = None,
  ) -> core.Learner:

    # The learner updates the parameters (and initializes them).
    return learning_lib.SGDLearner(
        network=networks,
        random_key=random_key,
        optimizer=optax.chain(
            optax.clip_by_global_norm(self._config.max_gradient_norm),
            optax.adam(self._config.learning_rate, eps=1e-3),
        ),
        target_update_period=self._config.target_update_period,
        data_iterator=dataset,
        loss_fn=self.loss_fn,
        replay_client=replay_client,
        replay_table_name=self._config.replay_table_name,
        counter=counter,
        num_sgd_steps_per_step=self._config.num_sgd_steps_per_step,
        logger=self._logger_fn())
```

**Comment by ethanluoyc** [comment]
This looks alright. Have you checked the logs? The coordinator should print out the number of steps seen so far. If not something is probably wrong.

**Comment by nikolamomchev** [comment]
@ethanluoyc thanks a lot for helping debug this!

@wcarvalho I don't believe we've experienced this problem before. Did you manage to resolve it?
Could you try the following things to help debug this:
* Can check if you can reproduce this with the standard [DistributedPPO example](https://github.com/deepmind/acme/blob/master/examples/gym/lp_ppo_jax.py). This uses the same layout, so if it works, then the problem is not in `DistributedLayout`.
* Can you make sure that the `max_number_of_steps` is properly propagated to the `DistributedLayout` constructor? I noticed that neither the [R2D2 example](https://github.com/deepmind/acme/blob/master/examples/atari/lp_r2d2_jax.py) nor the [R2D2 Agent](https://github.com/deepmind/acme/blob/076e8e1c1b8e13e8aae9708e94d3e2dca4a7cd03/acme/agents/jax/r2d2/agents.py#L38) are passing `max_number_of_steps` to the layout.

**Comment by wcarvalho** [comment]
I believe this was an error in my code. My apologies.
